src/compute/src/render/context.rs

```diff
@@ -550,14 +332,30 @@ where
     /// `flat_map` methods which allows this and can reduce the work done.
     pub fn as_collection(&self) -> (Collection<S, Row, Diff>, Collection<S, DataflowError, Diff>) {
         match &self {
-            ArrangementFlavor::Local(oks, errs) => (
-                oks.as_collection(move |borrow| SharedRow::pack(&**borrow)),
-                errs.as_collection(|k, &()| k.clone()),
-            ),
-            ArrangementFlavor::Trace(_, oks, errs) => (
-                oks.as_collection(move |borrow| SharedRow::pack(&**borrow)),
-                errs.as_collection(|k, &()| k.clone()),
-            ),
+            ArrangementFlavor::Local(oks, errs) => {
+                let mut datums = DatumVec::new();
+                (
+                    oks.as_collection(move |k, v| {
+                        let mut datums_borrow = datums.borrow();
+                        datums_borrow.extend(k.to_datum_iter());
+                        datums_borrow.extend(v.to_datum_iter());
+                        SharedRow::pack(&**datums_borrow)
+                    }),
```

Comment:
I can't really tell why this got more complicated. I can believe it all ties out, but paused here to acknowledge an understanding gap.